core/domain/user_services_test.py

```diff
@@ -770,3 +770,128 @@ def utcnow(cls):
                 user_services.get_user_settings(self.viewer_id).last_logged_in,
                 previous_last_logged_in_datetime)
             self.logout()
+
+
+class LastExplorationEditedIntegrationTests(test_utils.GenericTestBase):
+    EXP_ID = 'exp'
+
+    def setUp(self):
+        """Create users for creating and editing exploration."""
+        super(LastExplorationEditedIntegrationTests, self).setUp()
+        self.signup(self.OWNER_EMAIL, self.OWNER_USERNAME)
+        self.owner_id = self.get_user_id_from_email(self.OWNER_EMAIL)
+        self.signup(self.EDITOR_EMAIL, self.EDITOR_USERNAME)
+        self.editor_id = self.get_user_id_from_email(self.EDITOR_EMAIL)
+
+        self.save_new_valid_exploration(
+            self.EXP_ID, self.owner_id, end_state_name='End')
+
+    def edit_exploration(self):
+        """Edit an existing exploration as EDITOR."""
+        self.login(self.EDITOR_EMAIL)
+        exp_services.update_exploration(self.editor_id, self.EXP_ID, [{
+            'cmd': 'edit_exploration_property',
+            'property_name': 'objective',
+            'new_value': 'the objective'
+        }], 'Test edit')
+        self.logout()
+
+    def test_legacy_user(self):
+        """Test the case of a user who are editing exploration for first time
+        after the last edited time check was introduced.
+        """
+        # Set up a 'previous-generation' user.
+        user_settings = user_services.get_user_settings(self.editor_id)
+        user_settings.last_edited_an_exploration = None
+        user_services._save_user_settings(user_settings)  # pylint: disable=protected-access
+
+        self.assertIsNone(
+            user_services.get_user_settings(self.editor_id).
+            last_edited_an_exploration)
+
+        self.edit_exploration()
+
+        self.assertIsNotNone(
+            user_services.get_user_settings(self.editor_id).
+            last_edited_an_exploration)
+
+    def test_last_exp_edit_time_gets_updated(self):
+        self.edit_exploration()
+
+        # Decrease last exploration created time by 13 hours.
+        user_settings = user_services.get_user_settings(self.editor_id)
+        user_settings.last_edited_an_exploration = (
+            user_settings.last_edited_an_exploration -
+            datetime.timedelta(hours=13))
+        user_services._save_user_settings(user_settings) # pylint: disable=protected-access
+
+        previous_last_edited_an_exploration = (
+            user_services.get_user_settings(self.editor_id).
+            last_edited_an_exploration)
+        self.assertIsNotNone(previous_last_edited_an_exploration)
+
+        # Editing exploration after 13 time.
+        self.edit_exploration()
```

Comment:
Actually, let's drop the edit_exploration() method, and have update_exploration() inlined within here; ditto for the others. The reason is because self.edit_exploration() makes it sound like the instance of the test class is editing the exploration directly. For other similarly-named things we pass in user_id etc., but this is such a simple one-step operation (the login/logout can even be dropped) that I think we should just inline the call to update_exploration().

Also, that way, you can see what edits are actually being made -- for example, in this test, the same edit is being made twice.